### ISSUE TITLE
Problem with use of peek in NonNegativeNumber.rho

### ISSUE THREAD

**Issue by dckc** [rchain/rchain#3200]
https://github.com/rchain/rchain/blob/0fb0c616dadb553b7b1bea09e448e7a0e1ff9765/casper/src/main/resources/NonNegativeNumber.rho#L48-L50

**Comment by tgrospic** [comment]
@dckc Good catch! Unfortunately this is not by accident. In resolving of conflicts PR #2837 with @Isaac-DeFrain, we discovered strange errors when peek is used on one of two places in NNN contract. Error is manifesting as timeout exception in PoS Rholang tests. All integration tests are passing.

After each bug I resolved in RSpace I tried if it also resolves this bug. Here is the last try after joins fix (PR #3000) with some additional tests https://github.com/tgrospic/rchain/pull/10.
